tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/PortalServerLaunchConfigDelegate.java

```diff
@@ -76,6 +102,80 @@ public void launch(ILaunchConfiguration config, String mode, ILaunch launch, IPr
 		}
 	}
 
+	private static int _compareJavaVersions(IVMInstall vm, String ver) {
+		if (vm instanceof AbstractVMInstall) {
+			AbstractVMInstall install = (AbstractVMInstall)vm;
+
+			String vmver = install.getJavaVersion();
+
+			if (vmver == null) {
+				return -1;
+			}
+
+			// versionToJdkLevel only handles 3 char versions = 1.5, 1.6, 1.7, etc
+
+			if (vmver.length() > 3) {
+				vmver = vmver.substring(0, 3);
+			}
+
+			return Long.compare(CompilerOptions.versionToJdkLevel(vmver), CompilerOptions.versionToJdkLevel(ver));
+		}
+
+		return -1;
+	}
+
+	private static String _getJavaagentLocation() {
+		return LaunchingPlugin.getFileInPlugin(new Path("lib/javaagent-shaded.jar")).getAbsolutePath();
+	}
+
+	private static String _getJavaagentString() {
```

Comment:
This is introducing our own logic and we should be reusing the logic of the platform support.  We don't want to duplicate this code because we won't get future bug fixes.  We should just enable the advanced source lookup just like this do in WTP tomcat adapter (in latest WTP source).

But we can't directy call `allowAdvancedSourcelookup()` because that is only available in latest WTP.  We should call that by reflection if that method exists.